Tradução: intro.qmd (#44)
cienciadedatos/pt-r4ds: intro.qmd

# Introdução {#sec-intro}

```{r}
#| echo: false

source("_common.R")
```

> Ciência de dados é uma área emocionante que permite que pessoas desenvolvedoras transformem dados brutos em compreensão, insights e conhecimento.

Author: ```suggestion
Ciência de dados é uma área emocionante que permite que pessoas desenvolvedoras transformem dados brutos em compreensão, *insights* e conhecimento.
```

> Ciência de dados é uma área emocionante que permite que pessoas desenvolvedoras transformem dados brutos em compreensão, insights e conhecimento.

Author: Sobre a terceira pessoa, encontrei em dois lugares e corrigi! Obrigada!

> Ciência de dados é uma área emocionante que permite que pessoas desenvolvedoras transformem dados brutos em compreensão, *insights* e conhecimento.

Author: ```suggestion
Ciência de dados é uma área emocionante que permite que você transforme dados brutos em compreensão, *insights* e conhecimento.
```

> O objetivo do livro "R para Ciência de Dados" é ajudar você a aprender as ferramentas mais importantes em R que permitirão que pessoas desenvolvedoras realizem ciência de dados de forma eficiente e reproduzível, e se divirtam ao longo do caminho 😃.

Author: ```suggestion
O objetivo do livro "R para Ciência de Dados" é ajudar você a aprender as ferramentas mais importantes em R que permitirão que pessoas desenvolvedoras realizem ciência de dados de forma eficiente e reprodutível, e se divirtam ao longo do caminho 😃.
```

> O objetivo do livro "R para Ciência de Dados" é ajudar você a aprender as ferramentas mais importantes em R que permitirão que pessoas desenvolvedoras realizem ciência de dados de forma eficiente e reprodutível, e se divirtam ao longo do caminho 😃.

Author: ```suggestion
O objetivo do livro "R para Ciência de Dados" é ajudar você a aprender as ferramentas mais importantes em R que permitirão que você realize ciência de dados de forma eficiente e reprodutível, e se divirta ao longo do caminho 😃.
```

Após a leitura deste livro, você terá as ferramentas necessárias para enfrentar uma ampla variedade de desafios de ciência de dados usando as melhores partes do R.

## O que você aprenderá

A ciência de dados é um campo vasto, e não é possível dominá-la lendo apenas um único livro.
Este livro tem como objetivo fornecer a você uma base sólida nas ferramentas mais importantes e conhecimento suficiente para encontrar os recursos necessários para aprender mais quando for preciso.
Nosso modelo das etapas de um projeto típico de ciência de dados se parece com @fig-ds-diagram.

```{r}
#| label: fig-ds-diagram
#| echo: false
#| fig-cap: |

> #|   Em nosso modelo do ciclo de ciência de dados, você começa com a importação

Author: ```suggestion
#|   Em nosso modelo do processo de ciência de dados, você começa com a importação
```

> #|   Em nosso modelo do ciclo de ciência de dados, você começa com a importação

Author: Faz sentido! Acho que é pq estou muito acostumada a ouvir "ciclo da ciencia de dados" já.

#|   e organização dos dados. Em seguida, você compreende seus dados com um ciclo iterativo de
#|   transformação, visualização e modelagem. Você finaliza o ciclo
#|   comunicando seus resultados para outras pessoas.
#| fig-alt: |
#|   Um diagrama que exibe o ciclo de ciência de dados: Importar -> Organizar -> Compreender
#|   (com as fases Transformar -> Visualizar -> Modelar em um ciclo) ->
#|   Comunicar. Em torno de todas essas etapas está a etapa de Comunicar.
#| out.width: NULL

knitr::include_graphics("diagrams/data-science/base.png", dpi = 270)
```

Primeiro, você deve **importar** seus dados para o R.

> Isso geralmente significa que você pega dados armazenados em um arquivo, um banco de dados ou uma interface de programação de aplicação (*API*) da web e importa em uma base de dados no R.

Author: ```suggestion
Isso geralmente significa que você pega dados armazenados em um arquivo, um banco de dados ou uma *API* (interface de programação de aplicação, ou ***A**pplication **P**rogramming **I**nterface* em inglês) da web e importa em uma base de dados no R.
```

Author: @halian-vilela o que acha dessa ideia?

Reviewer: Acho ótimo! Creio que podemos ir nessa linha, caso fique muito grande ou a gente sinta que atrapalha o fluxo de leitura, aí a gente pensa em extrair para uma tooltip, se possível!

Author: tá bom!

> Isso geralmente significa que você pega dados armazenados em um arquivo, um banco de dados ou uma interface de programação de aplicação (*API*) da web e importa em uma base de dados no R.

Author: Coloquei essa sua última frase na wiki, mas não deixei exemplo ainda - esperando a gente decidir qual forma deixaremos!

> Isso geralmente significa que você pega dados armazenados em um arquivo, um banco de dados ou uma *API* (interface de programação de aplicação, ou ***A**pplication **P**rogramming **I**nterface* em inglês) da web e importa em uma base de dados no R.

Author: ```suggestion
Isso geralmente significa que você pega dados armazenados em um arquivo, um banco de dados ou uma *API* (interface de programação de aplicação, ou ***A**pplication **P**rogramming **I**nterface* em inglês) da web e importa em uma tabela (*data frame*) no R.
```

Se você não conseguir importar seus dados para o R, não poderá fazer ciência de dados com eles!

> Depois de importar seus dados, é uma boa ideia **organizá-los** (torná-los tidy).  

Author: ```suggestion
Depois de importar seus dados, é uma boa ideia **organizá-los** .  
```

> Depois de importar seus dados, é uma boa ideia **organizá-los** (torná-los tidy).  

Author: Boa ideia! fiz uma sugestão.

Organizar seus dados significa armazená-los em uma forma consistente que corresponda à semântica do conjunto de dados com a forma como ele é armazenado.

> Em resumo, quando seus dados estão organizados de forma *tidy*, cada coluna é uma variável e cada linha é uma observação.

Author: ```suggestion
Em resumo, quando seus dados estão organizados de forma *tidy*[^intro-nota-traducao-1], cada coluna é uma variável e cada linha é uma observação.

[^intro-nota-traducao-1]: **Nota de tradução**:  *tidy* é tanto um verbo em inglês que quer dizer "arrumar/organizar", quanto um paradigma de estruturação de dados. Caso queira saber mais, o artigo a seguir aprofunda este conceito: [WICKHAM, H. . Tidy Data. **Journal of Statistical Software**, [S. l.], v. 59, n. 10, p. 1–23, 2014. DOI: 10.18637/jss.v059.i10. Disponível em: https://www.jstatsoft.org/index.php/jss/article/view/v059i10. Acesso em: 4 nov. 2023.](https://www.jstatsoft.org/article/view/v059i10)
```

@albersonmiranda o que acha dessa nota?

> Dados organizados (*tidy data*) são importantes porque a estrutura consistente permite que você concentre seus esforços em responder perguntas sobre os dados, em vez de lutar para colocar os dados na forma correta para diferentes funções.

Author: ```suggestion
Dados organizados (*tidy data*) são importantes porque a estrutura consistente permite que você concentre seus esforços em responder perguntas sobre os dados, em vez de lutar para colocar os dados na forma correta para usar diferentes funções.
```

Dessa forma, quando você importar e organizar seus próprios dados, sua motivação permanecerá alta, porque você sabe que o esforço vale a pena.

Dentro de cada capítulo, procuramos seguir um padrão consistente: começar com alguns exemplos motivadores para que você possa entender o panorama geral e, em seguida, aprofundar nos detalhes.
Cada seção do livro é acompanhada de exercícios para ajudar você a praticar o que aprendeu.
Embora possa ser tentador pular os exercícios, a melhor maneira de aprender é praticando com problemas reais.

## O que você não aprenderá

Existem vários tópicos importantes que este livro não aborda.
Acreditamos que é importante manter um foco rigoroso no essencial para que você possa começar o mais rápido possível.
Isso significa que não é possível, neste livro, abordar todos os tópicos importantes.

### Modelagem

A modelagem é extremamente importante para a ciência de dados, mas é um tópico amplo e, infelizmente, não temos espaço suficiente para abordá-lo adequadamente aqui.
Para aprender mais sobre modelagem, recomendamos fortemente o livro [Tidy Modeling with R](https://www.tmwr.org), escrito por nossos colegas Max Kuhn e Julia Silge.

> Este livro ensinará a você a família de pacotes tidymodels, que, como você pode imaginar pelo nome, compartilha muitas convenções com os pacotes do tidyverse que usamos neste livro.

Author: Como você sugere para alterar? A ideia abaixo funciona?
```suggestion
O livro [Tidy Modeling with R](https://www.tmwr.org) ensinará a você a família de pacotes tidymodels, que, como você pode imaginar pelo nome, compartilha muitas convenções com os pacotes do tidyverse que usamos neste livro.
```

### Big data

Este livro orgulhosamente e principalmente foca em conjuntos de dados pequenos e que cabem na memória (*in-memory*).
Este é o lugar certo para começar, porque você não poderá lidar com *big data* a menos que já tenha experiência com bases de dados pequenas.
As ferramentas que você aprenderá ao longo da maior parte deste livro lidarão facilmente com centenas de megabytes de dados e, com um pouco de cuidado, você geralmente poderá usá-las para trabalhar com alguns gigabytes de dados.
Também mostraremos como obter dados de bancos de dados e arquivos parquet, ambos frequentemente usados para armazenar *big data*.

> Você não necessariamente conseguirá trabalhar com o conjunto de dados inteiro, mas isso não é um problema, pois você só precisa de um subconjunto ou uma amostra para responder à pergunta que te interessa.

Author: Essa versão que você citou é a primeira edição do livro!

A segunda edição é essa aqui:
https://r4ds.hadley.nz/intro

Sobre estar errado: acho que eles quiseram dizer que não é necessário todas as linhas e colunas. Se você for fazer um sumário, muitas vezes a gente usa apenas algumas colunas da base, né?
```suggestion
Você não necessariamente conseguirá trabalhar com o conjunto de dados inteiro, mas isso nem sempre é um problema, pois, em muitos casos, você só precisa de um subconjunto ou uma amostra para responder à pergunta que te interessa.
```

### Python, Julia e outros

Neste livro, você não aprenderá nada sobre Python, Julia ou qualquer outra linguagem de programação útil para a ciência de dados.
Isso não é porque achamos que essas ferramentas são ruins.
Elas não são!
E, na prática, a maioria das equipes de ciência de dados usa uma combinação de linguagens, muitas vezes pelo menos R e Python.
Mas acreditamos firmemente que é melhor dominar uma ferramenta de cada vez, e R é um ótimo ponto de partida.

## Pré-requisitos

Fizemos algumas suposições sobre o que você já sabe para aproveitar ao máximo este livro.
Você deve ter uma compreensão geral de matemática e é útil se você já tiver alguma experiência básica em programação.
Se você nunca programou antes, pode achar o livro [Hands on Programming with R](https://rstudio-education.github.io/hopr/), escrito por Garrett, um recurso valioso para complementar este livro.

Você precisará de quatro coisas para executar o código deste livro: R, RStudio, um conjunto de pacotes R chamado **tidyverse** e alguns outros pacotes.

> Pacotes são as unidades fundamentais de código R reproduzível.

Author: ```suggestion
Pacotes são as unidades fundamentais de código R reprodutível.
```

> Você precisará de quatro coisas para executar o código deste livro: R, RStudio, um conjunto de pacotes R chamado **tidyverse** e alguns outros pacotes.

Author: ```suggestion
Você precisará de quatro coisas para executar os códigos deste livro: R, RStudio, um conjunto de pacotes R chamado **tidyverse** e alguns outros pacotes.
```

Eles incluem funções reutilizáveis, documentação que descreve como usá-los e dados de exemplo.

### R

Para baixar o R, acesse o CRAN, o **c**omprehensive **R** **a**rchive **n**etwork, em <https://cloud.r-project.org>.
Uma nova versão principal do R é lançada anualmente, e há 2-3 lançamentos menores a cada ano.
É uma boa ideia atualizar regularmente.
A atualização pode ser um pouco complicada, especialmente para as versões principais que exigem que você reinstale todos os seus pacotes, mas adiá-la só torna as coisas piores.
Recomendamos o R 4.2.0 ou posterior para este livro.

### RStudio

> O RStudio é um ambiente de desenvolvimento integrado, ou IDE, para programação em R, que você pode baixar em <https://posit.co/download/rstudio-desktop/>.

Reviewer: ```suggestion
O RStudio é uma *IDE* (ambiente de desenvolvimento integrado), para programação em R, que você pode baixar em <https://posit.co/download/rstudio-desktop/>.
```

O RStudio é atualizado algumas vezes por ano e ele o informará automaticamente quando uma nova versão estiver disponível, portanto, não é necessário verificar periodicamente.
É uma boa ideia atualizar regularmente para aproveitar os recursos mais recentes e melhorados.
Para este livro, certifique-se de ter pelo menos o RStudio 2022.02.0.

Quando você inicia o RStudio, @fig-rstudio-console, você verá duas partes principais na interface: o painel de console (*Console*) e o painel de saída (*Output*).
Por enquanto, tudo o que você precisa saber é que você digita o código R no *Console* e pressiona Enter para executá-lo.
Você aprenderá mais à medida que avançarmos![^intro-1]

[^intro-1]: Se você deseja uma visão abrangente de todos os recursos do RStudio, consulte o Guia de uso do RStudio em <https://docs.posit.co/ide/user>.

```{r}
#| label: fig-rstudio-console
#| echo: false
#| out-width: ~
#| fig-cap: |

> #|   O IDE do RStudio possui duas partes principais: digite o código em R no Console

Reviewer: ```suggestion
#|   O RStudio possui duas partes principais: digite o código em R no Console
```
Acho que não precisa mais ficar citando **IDE**

#|   à esquerda e procure o painel Plots dentro do painel Output à direita.
#| fig-alt: |

> #|   O IDE do RStudio com os painéis Console e Output destacados.

Reviewer: ```suggestion
#|   O RStudio com os painéis Console e Output destacados.
```

[^intro-nota-traducao-1]: **Nota de tradução**:  *tidy* é tanto um verbo em inglês que quer dizer "arrumar/organizar", quanto um paradigma de estruturação de dados. Caso queira saber mais, o artigo a seguir aprofunda este conceito: [WICKHAM, H. . Tidy Data. **Journal of Statistical Software**, [S. l.], v. 59, n. 10, p. 1–23, 2014. DOI: 10.18637/jss.v059.i10. Disponível em: https://www.jstatsoft.org/index.php/jss/article/view/v059i10. Acesso em: 4 nov. 2023.](https://www.jstatsoft.org/article/view/v059i10)
Dados organizados (*tidy data*) são importantes porque a estrutura consistente permite que você concentre seus esforços em responder perguntas sobre os dados, em vez de lutar para colocar os dados na forma correta para usar diferentes funções.

Depois de ter dados organizados, normalmente o próximo passo é **transformá-los**.
A transformação inclui focar em observações de interesse (como todas as pessoas em uma cidade ou todos os dados do último ano), criar novas variáveis que são funções de variáveis existentes (como calcular a velocidade a partir da distância e do tempo) e calcular um conjunto de estatísticas resumidas (como contagens ou médias).

> Juntos, organizar e transformar são chamados de **manipulação de dados** (em inglês: *data wrangling*) porque colocar seus dados em uma forma natural de trabalhar frequentemente parece uma luta! (*wrangle*)

Author: ```suggestion
Juntos, organizar e transformar são chamados de **manipulação de dados**[^intro-nota-traducao-2]. 

[^intro-nota-traducao-2]**Nota de tradução**: Manipulação de dados é chamado em inglês de *data wrangling*, porque colocar seus dados em uma forma natural de trabalhar frequentemente parece uma luta (*wrangle*)! 
```

#|   comunicando seus resultados para outras pessoas.
#| fig-alt: |
#|   Um diagrama que exibe o ciclo de ciência de dados: Importar -> Organizar (Tidy) -> Compreender
#|   (com as fases Transformar -> Visualizar -> Modelar em um ciclo) ->
#|   Comunicar. Em torno de todas essas etapas está a etapa de Comunicar.
#| out.width: NULL

knitr::include_graphics("diagrams/data-science/base.png", dpi = 270)
```

Primeiro, você deve **importar** seus dados para o R.
Isso geralmente significa que você pega dados armazenados em um arquivo, um banco de dados ou uma *API* (interface de programação de aplicação, ou ***A**pplication **P**rogramming **I**nterface* em inglês) da web e importa em uma tabela (*data frame*) no R.
Se você não conseguir importar seus dados para o R, não poderá fazer ciência de dados com eles!

Depois de importar seus dados, é uma boa ideia **organizá-los** .  
Organizar seus dados significa armazená-los em uma forma consistente que corresponda à semântica do conjunto de dados com a forma como ele é armazenado.
Em resumo, quando seus dados estão organizados de forma *tidy*[^intro-nota-traducao-1], cada coluna é uma variável e cada linha é uma observação.

> [^intro-nota-traducao-1]: **Nota de tradução**:  *tidy* é tanto um verbo em inglês que quer dizer "arrumar/organizar", quanto um paradigma de estruturação de dados. Caso queira saber mais, o artigo a seguir aprofunda este conceito: [WICKHAM, H. . Tidy Data. **Journal of Statistical Software**, [S. l.], v. 59, n. 10, p. 1–23, 2014. DOI: 10.18637/jss.v059.i10. Disponível em: https://www.jstatsoft.org/index.php/jss/article/view/v059i10. Acesso em: 4 nov. 2023.](https://www.jstatsoft.org/article/view/v059i10)

Author: ```suggestion
[^intro-nota-traducao-1]**Nota de tradução**:  *tidy* é tanto um verbo em inglês que quer dizer "arrumar/organizar", quanto um paradigma de estruturação de dados. Caso queira saber mais, o artigo a seguir aprofunda este conceito: [WICKHAM, H. . Tidy Data. **Journal of Statistical Software**, [S. l.], v. 59, n. 10, p. 1–23, 2014. DOI: 10.18637/jss.v059.i10. Disponível em: https://www.jstatsoft.org/index.php/jss/article/view/v059i10. Acesso em: 4 nov. 2023.](https://www.jstatsoft.org/article/view/v059i10)
```

Cada seção do livro é acompanhada de exercícios para ajudar você a praticar o que aprendeu.
Embora possa ser tentador pular os exercícios, a melhor maneira de aprender é praticando com problemas reais.

## O que você não aprenderá

Existem vários tópicos importantes que este livro não aborda.
Acreditamos que é importante manter um foco rigoroso no essencial para que você possa começar o mais rápido possível.
Isso significa que não é possível, neste livro, abordar todos os tópicos importantes.

### Modelagem

A modelagem é extremamente importante para a ciência de dados, mas é um tópico amplo e, infelizmente, não temos espaço suficiente para abordá-lo adequadamente aqui.
Para aprender mais sobre modelagem, recomendamos fortemente o livro [Tidy Modeling with R](https://www.tmwr.org), escrito por nossos colegas Max Kuhn e Julia Silge.
O livro [Tidy Modeling with R](https://www.tmwr.org) ensinará a você a família de pacotes tidymodels, que, como você pode imaginar pelo nome, compartilha muitas convenções com os pacotes do tidyverse que usamos neste livro.

### Big data

> Este livro orgulhosamente e principalmente foca em conjuntos de dados pequenos e que cabem na memória (*in-memory*).

Author: ```suggestion
Este livro orgulhosamente e principalmente foca em conjuntos de dados pequenos e que cabem na memória[^intro-nota-traducao-3] (*in-memory*).

[^intro-nota-traducao-3]**Nota de tradução**: "Caber na memória" se refere à memória RAM (*random access memory*) do computador, cuja função é guardar temporariamente toda a informação que o computador precisa (por exemplo, as bases de dados importadas).
```

@albersonmiranda o que acha?

Você pode instalar o tidyverse completo com uma única linha de código:

```{r}
#| eval: false

install.packages("tidyverse")
```

Em seu computador, digite essa linha de código no console e pressione Enter para executá-la.
O R fará o download dos pacotes do CRAN e os instalará em seu computador.

Você não poderá usar as funções, objetos ou arquivos de ajuda de um pacote até carregá-lo com `library()`.
Depois de instalar um pacote, você pode carregá-lo usando a função `library()`:

```{r}
library(tidyverse)
```

Isso informa que o tidyverse carrega nove pacotes: dplyr, forcats, ggplot2, lubridate, purrr, readr, stringr, tibble, tidyr.
Esses são considerados o **núcleo** do tidyverse porque você os usará em quase todas as análises.

Os pacotes do tidyverse mudam com bastante frequência.
Você pode verificar se há atualizações disponíveis executando `tidyverse_update()`.

### Outros pacotes

Existem muitos outros pacotes excelentes que não fazem parte do tidyverse porque resolvem problemas em um domínio diferente ou são projetados com um conjunto diferente de princípios subjacentes.
Isso não os torna melhores ou piores; apenas os torna diferentes.

> Em outras palavras, o complemento ao tidyverse não é o messyverse, mas muitos outros universos de pacotes inter-relacionados.

Author: ```suggestion
Em outras palavras, o complemento ao tidyverse não é o messyverse[^intro-nota-traducao-4], mas muitos outros universos de pacotes inter-relacionados.

[^intro-nota-traducao-4] **Nota de tradução**: *tidyverse* é a união das palavras *tidy* (arrumado) e *universe* (universo), sendo então a ideia de um "universo arrumado". *messy* quer dizer desarrumado, e *messyverse* seria a ideia de um universo desarrumado. 
```

#> Error in library(ggrepel) : there is no package called ‘ggrepel’
```

Você precisará executar `install.packages("ggrepel")` para instalar o pacote.

## Executando código em R

A seção anterior mostrou vários exemplos de execução de código em R.
O código no livro parece assim:

```{r}
#| eval: true
1 + 2
```

Se você executar o mesmo código no seu *console*, ele parecerá assim:

```         
> 1 + 2
[1] 3
```

Existem duas diferenças principais.
No seu *console*, você digita após o `>`, chamado de **prompt**; não mostramos o prompt no livro.
No livro, a saída é comentada com `#>`; no seu *console*, ela aparece diretamente após o código.
Essas duas diferenças significam que se você estiver trabalhando com uma versão online do livro, poderá copiar facilmente o código do livro e colá-lo no *console*.

Ao longo do livro, usamos um conjunto consistente de convenções para se referir ao código:

-   Funções são exibidas em uma fonte de código e seguidas por parênteses, como `sum()` ou `mean()`.

> -   Outros objetos R (como dados ou argumentos de função) estão em uma fonte de código, sem parênteses, como `flights` ou `x`.

Author: ```suggestion
-   Outros objetos R (como dados ou argumentos de função) estão em uma fonte de código, sem parênteses, como `voos` ou `x`.
```

> -   Às vezes, para deixar claro de qual pacote um objeto vem, usaremos o nome do pacote seguido por quatro-pontos `::`, como `dplyr::mutate()` ou `nycflights13::flights`.

Author: ```suggestion
-   Às vezes, para deixar claro de qual pacote um objeto vem, usaremos o nome do pacote seguido por quatro-pontos `::`, como `dplyr::mutate()` ou `dados::voos`.
```

Usar base do pacote {dados}
